client/src/js/components/courseEvals/CourseEvals.js

```diff
@@ -36,169 +54,107 @@ const StyledSubHeading = styled("h2")`
 
-async function getCourseEvals (courseCode) {
+async function fetchAndSaveCourses () {
   try {
-    const res = await axios.get(wasetimeApiStatic.courseEvalsBaseURL + courseCode);
-    return res.data;
+    const res = await axios.get(wasetimeApiStatic.courseListAll);
+    const fetchedCourses = res.data.filter(course => course.term.includes("fall"));
+    const normalizedFetchedCourses = normalize(fetchedCourses, coursesSchema).entities.courses;
+    saveState({ fetchedCourses: normalizedFetchedCourses });
+    return normalizedFetchedCourses;
   } catch (error) {
     console.error(error);
   }
 }
 
-const getRelatedCourses = (fetchedCourses, courseCode, thisCourseKey) => {
-  // Todo: Get related courses with course code
-  // let relatedCourses = fetchedCourses.filter(c => c.course_code === courseCode && getCourseKey(c) !== thisCourseKey);
-
-  return [
-    {
-      "_id":"2603033019012020260303301926",
-      "course_code": 'INFP31ZL',
-      "year":2020,
-      "term":"fallSem",
-      "title":"Software Engineering B",
-      "title_jp":"ソフトウェア工学B",
-      "instructor":"HAGIMOTO, Junzo/HIRANABE, Kenji/FUKAZAWA, Yoshiaki/WASHIZAKI, Hironori",
-      "instructor_jp":"萩本 順三/平鍋 健児/深澤 良彰/鷲崎 弘宜",
-      "occurrences":[
-        {
-           "day":3,
-           "start_period":5,
-           "end_period":5,
-           "building":"-1",
-           "classroom":"undecided",
-           "location":"-1-undecided"
-        }
-      ],
-      "lang":"JP",
-      "keys":[
-        {
-           "school":"FSE",
-           "key":"2603033019012020260303301926"
-        }
-      ]
-    },
-    {
-      "_id":"26GC02300301202026MC03300926",
-      "course_code": 'INFP31ZL',
-      "year":2020,
-      "term":"fallSem",
-      "title":"Software Engineering",
-      "title_jp":"Software Engineering",
-      "instructor":"ARCAINI, Paolo/KOBAYASHI, Tsutomu/FUKAZAWA, Yoshiaki/HONIDEN, Shinichi/WASHIZAKI, Hironori",
-      "instructor_jp":"アルカイニ パオロ/小林 努/深澤 良彰/本位田 真一/鷲崎 弘宜",
-      "occurrences":[
-        {
-           "day":1,
-           "start_period":3,
-           "end_period":3,
-           "building":"-1",
-           "classroom":"undecided",
-           "location":"-1-undecided"
-        }
-      ],
-      "keywords":[
-        "IPSE"
-      ],
-      "lang":"EN",
-      "keys":[
-        {
-           "school":"FSE",
-           "key":"26GC02300301202026MC03300926"
-        }
-      ]
-    }
-  ];
+const getRelatedCourses = (loadedCourses, courseCode, thisCourseKey, thisCourseTitle) => {
+  const relatedCourseIDs = Object.keys(loadedCourses).filter(courseID => loadedCourses[courseID].code === courseCode && getCourseKey(loadedCourses[courseID]) !== thisCourseKey);
+  const relatedCourses = relatedCourseIDs.map(courseID => loadedCourses[courseID]);
+  const sortedRelatedCourses = relatedCourses.sort((a, b) => levenshtein(thisCourseTitle, a.title) - levenshtein(thisCourseTitle, b.title)).slice(0,3);
```

Comment:
It would be too slow to display all related courses especially for CJL. I remembered we conclude at displaying top several related courses only.

Reply:
Hmm, I think just displaying the courses is fine though? We're not calling any APIs except for the top 3 courses so it should be fast even if you have a lot of related courses. Could you help me look into a couple of CJL courses and see how many related courses they have? Thanks!

Reply:
Oh I got it. Display all related courses but just fetch evaluations for the top 3 ones right?
BTW I think there are more than a hundred of related courses for a CJL course

Reply:
Correct! Because it's called "related courses" so technically we should display all related courses. I think we agreed on this point with @AustinZhu. 

More than a hundred... hmm if that's the case perhaps we should add a limit and display it to the users as well. What about "Related courses (top 10)" (kind of ugly though lol) or other ways of displaying?

Reply:
I think displaying top 10 is fine!